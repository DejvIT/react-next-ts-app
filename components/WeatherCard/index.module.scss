@import '@app/styles/modules/colors';
@import '@app/styles/modules/_mixins.scss';

.card {
  background-color: var(--clrPrimaryLight) !important;
}
.cardSelected {
  background-color: var(--clrPrimaryDark) !important;
  box-shadow: 0px 5px 10px var(--clrDarkSecondary), 0px 3px 3px var(--clrDarkTertiary);
}
