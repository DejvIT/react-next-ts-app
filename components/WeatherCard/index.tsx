import classNames from 'classnames';
import React, { memo } from 'react';
import { Card } from 'react-bootstrap';

import Button from '@app/components/Button';
import { formatDateTime } from '@app/helpers';
import { WeatherUnits } from '@app/types';

import styles from './index.module.scss';

interface Props {
  time: string;
  airPressure?: number;
  airTemperature?: number;
  cloudAreaFraction?: number;
  relativeHumidity?: number;
  windDirection?: number;
  windSpeed?: number;
  units: WeatherUnits;
  onClick: () => void;
  variant: 'Primary' | 'Secondary' | 'Success' | 'Danger' | 'Warning' | 'Info' | 'Light' | 'Dark';
  selected?: boolean;
}

const WeatherCard: React.FC<Props> = ({
  time,
  airPressure,
  airTemperature,
  cloudAreaFraction,
  relativeHumidity,
  windDirection,
  windSpeed,
  units,
  variant,
  selected,
  onClick,
}) => {
  return (
    <Card
      bg={variant.toLowerCase()}
      className={classNames(selected ? styles.cardSelected : styles.card, 'm-4')}
      key={variant}
      style={{ width: '18rem' }}
      text={variant.toLowerCase() === 'light' ? 'dark' : 'white'}
    >
      <Card.Header>{formatDateTime(time)}</Card.Header>
      <Card.Body>
        <Card.Title>{variant} Card Title</Card.Title>
        <Card.Text>
          <div>
            Air pressure: {airPressure} {units.air_pressure_at_sea_level}
          </div>
          <div>
            Air temperature: {airTemperature} {units.air_temperature}
          </div>
          <div>Cloud area fraction: {cloudAreaFraction}%</div>
          <div>Relative humidity: {relativeHumidity}%</div>
          <div>Wind direction: {windDirection}Â°</div>
          <div>
            Wind speed: {windSpeed} {units.wind_speed}
          </div>
        </Card.Text>
      </Card.Body>
      <Card.Footer>
        <Button block onClick={onClick} size="large" variant="primary" withChevronRight>
          Open detail
        </Button>
      </Card.Footer>
    </Card>
  );
};

export default memo(WeatherCard);
