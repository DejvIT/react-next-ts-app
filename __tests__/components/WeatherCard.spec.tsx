import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import WeatherCard from "@app/components/WeatherCard";
import { unitsMockData, weatherCardMockData } from "@app/__tests__/mock-data/weather-mock";

const data = weatherCardMockData;

describe('WeatherCard component', () => {
  it('Should render component in primary variant', () => {
    const component = shallow(
        <WeatherCard
            airPressure={data?.instant?.details.air_pressure_at_sea_level}
             airTemperature={data?.instant?.details.air_temperature}
             cloudAreaFraction={data?.instant?.details.cloud_area_fraction}
             onClick={jest.fn()}
             relativeHumidity={data?.instant?.details.relative_humidity}
             time='2023-02-05T21:00:00Z'
             units={unitsMockData}
             variant='Primary'
             windDirection={data?.instant?.details.wind_from_direction}
             windSpeed={data?.instant?.details.wind_speed}
        />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });


  it('Should render component in info variant', () => {
    const component = shallow(
        <WeatherCard
            airPressure={data?.instant?.details.air_pressure_at_sea_level}
            airTemperature={data?.instant?.details.air_temperature}
            cloudAreaFraction={data?.instant?.details.cloud_area_fraction}
            onClick={jest.fn()}
            relativeHumidity={data?.instant?.details.relative_humidity}
            time='2023-02-05T21:00:00Z'
            units={unitsMockData}
            variant='Info'
            windDirection={data?.instant?.details.wind_from_direction}
            windSpeed={data?.instant?.details.wind_speed}
        />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });


  it('Should render component in selected primary variant', () => {
    const component = shallow(
        <WeatherCard
            airPressure={data?.instant?.details.air_pressure_at_sea_level}
            airTemperature={data?.instant?.details.air_temperature}
            cloudAreaFraction={data?.instant?.details.cloud_area_fraction}
            onClick={jest.fn()}
            relativeHumidity={data?.instant?.details.relative_humidity}
            time='2023-02-05T21:00:00Z'
            units={unitsMockData}
            selected
            variant='Primary'
            windDirection={data?.instant?.details.wind_from_direction}
            windSpeed={data?.instant?.details.wind_speed}
        />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
