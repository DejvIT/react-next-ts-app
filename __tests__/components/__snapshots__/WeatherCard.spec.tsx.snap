// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard component Should render component in info variant 1`] = `
<Card
  bg="info"
  body={false}
  className="card m-4"
  key="Info"
  style={
    Object {
      "width": "18rem",
    }
  }
  text="white"
>
  <CardHeader>
    5. 2. 2023 21:00
  </CardHeader>
  <CardBody>
    <CardTitle>
      Info
       Card Title
    </CardTitle>
    <CardText>
      <div>
        Air pressure: 
        1038.7
         
        hPa
      </div>
      <div>
        Air temperature: 
        -11.9
         
        celsius
      </div>
      <div>
        Cloud area fraction: 
        75
        %
      </div>
      <div>
        Relative humidity: 
        79.3
        %
      </div>
      <div>
        Wind direction: 
        1.2
        °
      </div>
      <div>
        Wind speed: 
        3.6
         
        m/s
      </div>
    </CardText>
  </CardBody>
  <CardFooter>
    <Memo(Button)
      block={true}
      onClick={[MockFunction]}
      size="large"
      variant="primary"
      withChevronRight={true}
    >
      Open detail
    </Memo(Button)>
  </CardFooter>
</Card>
`;

exports[`WeatherCard component Should render component in primary variant 1`] = `
<Card
  bg="primary"
  body={false}
  className="card m-4"
  key="Primary"
  style={
    Object {
      "width": "18rem",
    }
  }
  text="white"
>
  <CardHeader>
    5. 2. 2023 21:00
  </CardHeader>
  <CardBody>
    <CardTitle>
      Primary
       Card Title
    </CardTitle>
    <CardText>
      <div>
        Air pressure: 
        1038.7
         
        hPa
      </div>
      <div>
        Air temperature: 
        -11.9
         
        celsius
      </div>
      <div>
        Cloud area fraction: 
        75
        %
      </div>
      <div>
        Relative humidity: 
        79.3
        %
      </div>
      <div>
        Wind direction: 
        1.2
        °
      </div>
      <div>
        Wind speed: 
        3.6
         
        m/s
      </div>
    </CardText>
  </CardBody>
  <CardFooter>
    <Memo(Button)
      block={true}
      onClick={[MockFunction]}
      size="large"
      variant="primary"
      withChevronRight={true}
    >
      Open detail
    </Memo(Button)>
  </CardFooter>
</Card>
`;

exports[`WeatherCard component Should render component in selected primary variant 1`] = `
<Card
  bg="primary"
  body={false}
  className="cardSelected m-4"
  key="Primary"
  style={
    Object {
      "width": "18rem",
    }
  }
  text="white"
>
  <CardHeader>
    5. 2. 2023 21:00
  </CardHeader>
  <CardBody>
    <CardTitle>
      Primary
       Card Title
    </CardTitle>
    <CardText>
      <div>
        Air pressure: 
        1038.7
         
        hPa
      </div>
      <div>
        Air temperature: 
        -11.9
         
        celsius
      </div>
      <div>
        Cloud area fraction: 
        75
        %
      </div>
      <div>
        Relative humidity: 
        79.3
        %
      </div>
      <div>
        Wind direction: 
        1.2
        °
      </div>
      <div>
        Wind speed: 
        3.6
         
        m/s
      </div>
    </CardText>
  </CardBody>
  <CardFooter>
    <Memo(Button)
      block={true}
      onClick={[MockFunction]}
      size="large"
      variant="primary"
      withChevronRight={true}
    >
      Open detail
    </Memo(Button)>
  </CardFooter>
</Card>
`;
